src/rgw/rgw_redis_lock.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "rgw_redis_common.h"
+
+namespace rgw {
+namespace redislock {
+
+using boost::redis::config;
+using boost::redis::connection;
+
+// Needed if we implement tags and description for the lock
+// struct redis_lock {
+//   std::string name;
+//   std::string cookie;
+//   std::string description;
+//   std::string tag;
+// };
+
+int initLock(boost::asio::io_context& io, connection* conn, config* cfg,
+             optional_yield y);
+int lock(connection* conn, const std::string name, const std::string cookie,
```

Comment:
do we have a real "init" api in the case of the `cls_lock`, or, is the initiation always implicit wit hthe first lock?

Reply:
I have removed explicit `init` in both lock and queue. 
Assuming this should be loaded when either the Redis itself is started or every time a rgw spawns.

We discussed that the latter is preferable. 
Can you guide me on how to do that?

Reply:
* loading the lua scripts into redis should happen when RGW start

* we need "init_queue()" API to create a specific empty queue
* as well as "delete_queue()" to remove it
these APIs are unrelated to loading the scripts

Reply:
For Redis, an empty queue is automatically deleted and we cannot create an empty queue. 

What purpose will this serve? 
We can use a dummy entry at the beginning of all queues to achieve this if need be.

Reply:
look at the rgw_notify.cc exiting code.
we have an API to delete a queue. if a queue is deleted you must not read or write to it.
we should not implicitly create the queue on first write, and should not implicitly delete the queue when we did the last pop.

the indication that a queue exists or not (regardless of being empty) is important for our code to keep polling the queue.
since another client may perform the deletion, we cannot relly on being notified that a queue was deleted, except fro mthe fact that it is not in redis anymore